controllers/azurecluster_reconciler.go

```diff
@@ -101,6 +104,10 @@ func (s *azureClusterService) Reconcile(ctx context.Context) error {
 		return errors.Wrapf(err, "failed to reconcile route table")
 	}
 
+	if err := s.natGatewaySvc.Reconcile(ctx); err != nil {
```

Comment:
Shouldn't this be reconciled after public ip? With the current reconcile ordering, this will be conflicting because subnet is reconciled before public ip.

Reply:
I think you forgot to add reconcile for delete.

Reply:
Right, I re-ordered the services and added the delete reconciler.